src/js/apps/patients/schedule/reduced-schedule/reduced_schedule_app.js

```diff
@@ -97,19 +97,43 @@ export default App.extend({
     if (!this.isRestarting()) this.stopChildApp('filters');
   },
   beforeStart() {
-    const filter = this.getState().getEntityFilter();
+    this.filters = this.getState().getEntityFilter();
     const fields = { flows: ['name', 'state'] };
-    return Radio.request('entities', 'fetch:actions:collection', { data: { filter, fields } });
+
+    return Radio.request('entities', 'fetch:actions:collection', { data: { filter: this.filters, fields } });
   },
   onStart(options, collection) {
     this.collection = collection;
     this.filteredCollection = collection.clone();
 
+    this.subscribe();
+
     this.listenTo(this.filteredCollection, 'reset', this.showCountView);
     this.showCountView();
 
     this.showList();
   },
+  subscribe() {
+    const channel = Radio.channel('ws');
+
+    channel.request('subscribe', this.collection.models, { filters: { actions: this.filters } });
+
+    this.listenTo(channel, 'message', ({ category }, model) => {
+      if (category === 'ActionDueChanged') {
+        this.showList();
+        return;
+      }
+
+      if (this.collection.get(model) || model.isFetching()) return;
+
+      model.fetch().then(() => {
+        this.collection.add(model);
+        Radio.request('ws', 'add', model);
+
+        this.showList();
```

Comment:
I'm not sure how else to do this yet. 

When I test this locally, re-rendering the entire list doesn't seem to cause any issues. But I'm worried about what happens in production when the list is 100's of actions long.

Reply:
correct this doesn't seem like it will scale.. plus it will completely blow away any action the user is currently taking anywhere on the list